lib/core/src/sockComm.cpp

```diff
@@ -393,9 +385,23 @@ sockOpenForInConn( rsComm_t *rsComm, int *portNum, char **addr, int proto ) {
         return status;
     }
 
-    if ( addr != NULL ) {
-        *addr = ( char * )malloc( sizeof( char ) * LONG_NAME_LEN );
-        gethostname( *addr, LONG_NAME_LEN );
+    if (addr) {
+        namespace net = irods::experimental::net;
+
+        if (const auto hostname = net::hnc_lookup("localhost"); hostname) {
+            *addr = static_cast<char*>(std::malloc(sizeof(char) * hostname->size() + 1));
+            addr[hostname->size()] = 0;
+            rstrcpy(*addr, hostname->data(), LONG_NAME_LEN);
+        }
+        else {
+            *addr = static_cast<char*>(malloc(sizeof(char) * LONG_NAME_LEN));
+            gethostname(*addr, LONG_NAME_LEN);
+
+            const auto alias = resolve_hostname_from_hosts_config(*addr);
+            net::hnc_insert_or_assign("localhost", alias, std::chrono::seconds{20});
+
+            rstrcpy(*addr, alias.data(), LONG_NAME_LEN);
+        }
```

Comment:
do we see any need for the hostname cache here?

`resolve_hostname_from_hosts_config` will read the config file once for every agent. imo, there isn't much gained from using the hostname cache. maybe, this is a little faster given it is just a lookup once cached.

Reply:
on the server, yes?    anything to reduce network traffic is a speedup.

on the client... no.

Reply:
Well, that's the thing. I don't think this will affect network traffic. gethostname() is likely just reading a local file.

Reply:
On a different note, this code returns the first entry from the hosts_config.json.